sdk_container/src/third_party/coreos-overlay/sys-kernel/coreos-kernel/coreos-kernel-6.12.44.ebuild

```
@@ -89,6 +89,14 @@ src_compile() {
 
 	tc-export PKG_CONFIG
 	"${ESYSROOT}"/usr/bin/update-bootengine -k "${KV_FULL}" -o "${S}"/build/bootengine.cpio "${BE_ARGS[@]}" || die
+	mkdir "${S}"/build/bootengine
+	cat "${S}"/build/bootengine.cpio | while cpio --no-absolute-filenames -d -m -D "${S}"/build/bootengine -i ; do :; done
+	# Lacking read permissions even for user
+	chmod u+r "${S}"/build/bootengine/etc/gshadow
+	# CPU microcode should stay in the minimal initrd
+	rm -rf "${S}"/build/bootengine/kernel "${S}"/build/bootengine/early_cpio
+	# TODO: rework /usr verity mount to reuse the one from the minimal initrd
+	mksquashfs "${S}"/build/bootengine "${S}"/build/bootengine.img -all-root -noappend -xattrs-exclude ^btrfs.
```

```suggestion
	mkdir "${S}"/build/bootengine || die
	pushd "${S}"/build/bootengine || die
	sudo lsinitrd --unpack "${S}"/build/bootengine.cpio || die
	# TODO: rework /usr verity mount to reuse the one from the minimal initrd
	sudo mksquashfs . "${S}"/build/bootengine.img -noappend -xattrs-exclude ^btrfs. || die
	popd || die
```

This is running with `RESTRICT="userpriv"`, so you can use `sudo` here. That will ensure the device nodes are created correctly. `lsinitrd --unpack` will not include the microcode.

---

Also, might as well use `mkfs.erofs` now!

---

With bsdtar (which is in the SDK), you can even avoid extracting if `dracut` is called with `--no-compress --no-early-microcode`.

```sh
bsdtar -c @"${S}"/build/bootengine.cpio | mkfs.erofs --tar=f -zlz4hc "${S}"/build/bootengine.raw